src/adapters/git.ts

```diff
@@ -70,9 +70,7 @@ abstract class GitAdapter implements IRepoAdapter {
   }
 
   public async pushRepo(repo: IRepo, force: boolean): Promise<void> {
-    const options = {
-      '--force': force,
-    };
+    const options = force ? ['--force'] : undefined;
```

Comment:
I believe this is necessary because when I update `simple-git` from `v2.13.2` to `v2.31.0`, I get the following error on build:

```bash
src/adapters/git.ts:77:49 - error TS2345: Argument of type '{ '--force': boolean; }' is not assignable to parameter of type 'string[] | Record<string, string | number | null> | undefined'.
  Type '{ '--force': boolean; }' is not assignable to type 'Record<string, string | number | null>'.
    Property ''--force'' is incompatible with index signature.
      Type 'boolean' is not assignable to type 'string | number | null'.

     await this.git(repo).push('origin', 'HEAD', options);
```

I believe this originally stems from slightly incorrect usage of `simple-git` (or that package made a breaking change in a minor update). According to https://github.com/steveukx/git-js#options-as-an-object and the error above, options can be a string array, object or undefined. When an object is passed, key-value pairs should be `string: { string, number, null }`, _not_ `string: boolean`.

The change I propose here should be backwards-compatible while also meeting the expected usage of `simple-git`, but please let me know if not, or whether this should be broken out into its own PR.

Reply:
P.S., was thinking of `const options = !!force ? ['--force'] : undefined;` originally but it seemed redundant as `force` is a boolean. Since Typescript only kicks in at compile time though, is that type coercion advisable?

Reply:
I'd recommend splitting this out into a separate PR to mitigate risk and keep the focus on the feature.

If you're up for doing so, it might also be good to roll in the change for `joi` (see [1](https://github.com/NerdWalletOSS/shepherd/issues/354)).

References
1. https://github.com/NerdWalletOSS/shepherd/issues/354

Reply:
> P.S., was thinking of `const options = !!force ? ['--force'] : undefined;` originally but it seemed redundant as `force` is a boolean. Since Typescript only kicks in at compile time though, is that type coercion advisable?

What are alternatives to coercion?

Reply:
In my mind, coercion is a bit redundant, so the alternatives are to do coercion at all or not.  We could also set a sane default and do input validation for the force parameter.

Reply:
I reverted the `simple-git` update however, so closing this and will reference in a new PR.